core/src/main/java/com/linecorp/armeria/client/unsafe/DefaultPooledHttpClient.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.unsafe;
+
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.client.HttpClient;
+import com.linecorp.armeria.common.unsafe.PooledHttpRequest;
+import com.linecorp.armeria.common.unsafe.PooledHttpResponse;
+import com.linecorp.armeria.common.util.AbstractUnwrappable;
+
+final class DefaultPooledHttpClient extends AbstractUnwrappable<HttpClient> implements PooledHttpClient {
+
+    DefaultPooledHttpClient(HttpClient delegate) {
+        super(delegate);
+    }
+
+    @Override
+    public PooledHttpResponse execute(ClientRequestContext ctx, PooledHttpRequest req) throws Exception {
+        // Always a wrapped non-pooled client, make sure it gets a normal request.
+        assert !(delegate() instanceof PooledHttpClient);
```

Comment:
nit: How about moving this check to the constructor?

Reply:
I put it here more as documentation than anything so think it's better here. But if the comment is enough I can remove it?

Reply:
I don't mind. Fine with having it here.

Reply:
Don't think it matters so much - this is just a check against programming issue, the constructor is always called with something that isn't a `PooledHttpClient` My 20 days old comment still stands, I put it mainly as documentation ;)